src/simsoptpp/vec3dsimd.h

```diff
@@ -177,24 +200,371 @@ inline Vec3dSimd cross(int i, Vec3dSimd& b){
+        lhs.y += rhs.y;
+        lhs.z += rhs.z;
+        return lhs;
+    }
+
+    Vec3dStd& operator+=(const Vec3dStd& rhs) {
+        this->x += rhs.x;
+        this->y += rhs.y;
+        this->z += rhs.z;
+        return *this;
+    }
+
+    Vec3dStd& operator-=(const Vec3dStd& rhs) {
+        this->x -= rhs.x;
+        this->y -= rhs.y;
+        this->z -= rhs.z;
+        return *this;
+    }
+
+    Vec3dStd& operator*=(const double& rhs) {
+        this->x *= rhs;
+        this->y *= rhs;
+        this->z *= rhs;
+        return *this;
+    }
+
+    friend Vec3dStd operator-(Vec3dStd lhs, const Vec3d& rhs) {
+        lhs.x -= rhs[0];
+        lhs.y -= rhs[1];
+        lhs.z -= rhs[2];
+        return lhs;
+    }
+
+    friend Vec3dStd operator-(Vec3dStd lhs, const Vec3dStd& rhs) {
+        lhs.x -= rhs.x;
+        lhs.y -= rhs.y;
+        lhs.z -= rhs.z;
+        return lhs;
+    }
+
+    friend Vec3dStd operator*(Vec3dStd lhs, const double& rhs) {
+        lhs.x *= rhs;
+        lhs.y *= rhs;
+        lhs.z *= rhs;
+        return lhs;
+    }
+
+    friend Vec3dStd operator*( const double& lhs, Vec3dStd rhs) {
+        rhs.x *= lhs;
+        rhs.y *= lhs;
+        rhs.z *= lhs;
+        return rhs;
+    }
+
+};
+
+
+inline double inner(const Vec3dStd& a, const Vec3dStd& b){
+    return a.x * b.x + a.y * b.y + a.z * b.z;
 }
 
-inline simd_t normsq(Vec3dSimd& a){
-    return xsimd::fma(a.x, a.x, xsimd::fma(a.y, a.y, a.z*a.z));
+inline double inner(const Vec3d& b, const Vec3dStd& a){
+    return a.x *  b[0] + a.y * b[1] + a.z * b[2];
+}
+
+inline double inner(const Vec3dStd& a, const Vec3d& b){
+    return a.x * b[0] + a.y * b[1] + a.z * b[2];
+}
+
+inline double inner(int i, Vec3dStd& a){
+    switch(i){
+        case 0: return a.x;
+        case 1: return a.y;
+        case 2: return a.z;
+    };
+}
+
+inline Vec3dStd cross(Vec3dStd& a, Vec3dStd& b){
+    return Vec3dStd(
+            (a.y * b.z - a.z * b.y),
+            (a.z * b.x - a.x * b.z),
+            (a.x * b.y - a.y * b.x));
+}
+
+inline Vec3dStd cross(Vec3dStd& a, Vec3d& b){
+    return Vec3dStd(
+            (a.y * b[2] - a.z * b[1]),
+            (a.z * b[0] - a.x * b[2]),
+            (a.x * b[1] - a.y * b[0]));
+}
+
+inline Vec3dStd cross(Vec3d& a, Vec3dStd& b){
+    return Vec3dStd(
+            (a[1] * b.z - a[2] * b.y),
+            (a[2] * b.x - a[0] * b.z),
+            (a[0] * b.y - a[1] * b.x));
+}
+
+inline Vec3dStd cross(Vec3dStd& a, int i){
+    switch(i) {
+        case 0: return Vec3dStd(0., a.z, -a.y);
+        case 1: return Vec3dStd(-a.z, 0., a.x);
+        case 2: return Vec3dStd(a.y, -a.x, 0.);
+    }
+}
+
+inline Vec3dStd cross(int i, Vec3dStd& b){
+    switch(i){
+        case 0: return Vec3dStd(0., -b.z, b.y);
+        case 1: return Vec3dStd(b.z, 0., -b.x);
+        case 2: return Vec3dStd(-b.y, b.x, 0.);
+    }
+}
+
+inline double normsq(Vec3dStd& a){
+    return a.x * a.x + a.y * a.y + a.z * a.z;
+}
+
+struct alignas(ALIGN_BYTES) Vec3dSimdPortable1 {
```

Comment:
Admittedly, Vec3d was not documented previously, but now that there are two structures in use:
Vec3dstd and Vec3dPortable1, I think for readability the purpose of these two structures should be documented.  Why do we need both?

Reply:
I removed Vec3dPortable1 since it is not used in production. And I added comments for Vec3dStd.